File: samples/dotnet/kernel-syntax-examples/Example02_Pipeline.cs
// Copyright (c) Microsoft. All rights reserved.

using System;
using System.Threading.Tasks;
using Microsoft.Extensions.Logging;
using Microsoft.SemanticKernel;
using Microsoft.SemanticKernel.CoreSkills;
using Microsoft.SemanticKernel.Orchestration;
using RepoUtils;

// ReSharper disable once InconsistentNaming
public static class Example02_Pipeline
{
    private static readonly ILogger s_log = ConsoleLogger.Log;

    public static async Task RunAsync()
    {
        Console.WriteLine("======== Pipeline ========");

        IKernel kernel = new KernelBuilder().WithLogger(s_log).Build();

        // Load native skill
        var text = kernel.ImportSkill(new TextSkill());

        SKContext result = await kernel.RunAsync("    i n f i n i t e     s p a c e     ",
            text["TrimStart"],
            text["TrimEnd"],
            text["Uppercase"]);

        Console.WriteLine(result);
    }
}

File: java/samples/sample-code/src/main/java/com/microsoft/semantickernel/syntaxexamples/Example02_Pipeline.java
package com.microsoft.semantickernel.syntaxexamples;

import com.microsoft.semantickernel.Kernel;
import com.microsoft.semantickernel.KernelConfig;
import com.microsoft.semantickernel.builders.SKBuilders;
import com.microsoft.semantickernel.coreskills.TextSkill;
import com.microsoft.semantickernel.orchestration.SKContext;
import com.microsoft.semantickernel.skilldefinition.ReadOnlyFunctionCollection;
import reactor.core.publisher.Mono;

public class Example02_Pipeline {
    public static void main(String[] args) {
        KernelConfig kernelConfig = SKBuilders.kernelConfig().build();
        Kernel kernel = SKBuilders.kernel().withKernelConfig(kernelConfig).build();

        // Load native skill
        ReadOnlyFunctionCollection text = kernel.importSkill(new TextSkill(), null);

        Mono<SKContext> result =
                kernel.runAsync(
                        "    i n f i n i t e     s p a c e     ",
                        text.getFunction("LStrip"),
                        text.getFunction("RStrip"),
                        text.getFunction("Uppercase"));

        System.out.println(result.block().getResult());
    }
}
